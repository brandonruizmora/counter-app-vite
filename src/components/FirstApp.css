html, body {
    color: #fff;
    background-color: rgb(26, 30, 50);
}

.container {
    align-items: center;
    display: flex;
    height: 95vh;
    justify-content: center;
}

.center-text {
    text-align: center;
}

button {
    border-radius: 10px;
    font-size: 1rem;
    font-weight: bold;
    margin-left: 10px;
    margin-right: 10px;
    padding: 10px;
}