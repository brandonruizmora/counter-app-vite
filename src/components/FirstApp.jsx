
import PropTypes from 'prop-types';
import { useState } from 'react';

const FirstApp = ({ value }) => {

  const [counter, setCounter] = useState(value);

  const handleClickAdd = function () {
    setCounter(counter + 1);
  }

  const handleClickSubstract = function () {
    setCounter(counter - 1);
  }

  const handleClickReset = function () {
    setCounter(value);
  }

  return (
    <div className='container'>
      <div className='center-text'>
        <h1>CounterApp</h1>
        <h2 data-testid="counter">{counter}</h2>
        <button type='button' onClick={handleClickAdd}>+1</button>
        <button type='button' onClick={handleClickSubstract}>-1</button>
        <button aria-label='btn-reset' type='button' onClick={handleClickReset}>reset</button>
      </div>
    </div>
  )
}

FirstApp.propTypes = {
  value: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
  value: 0
}

export default FirstApp;