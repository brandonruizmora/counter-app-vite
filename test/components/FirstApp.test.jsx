import { render, screen, fireEvent } from "@testing-library/react";
import FirstApp from "../../src/components/FirstApp";

describe('Test suit First App', () => {

    test('Debe hacer match con el snapshot estado default', () => {
        const { container } = render(<FirstApp />)
        expect(container).toMatchSnapshot();
    });

    test('Debe mostrar el contador inicial en 100', () => {
        render(<FirstApp value={100} />);
        // screen.debug(); // --> console log del render
        expect(screen.getByTestId('counter')).toBeTruthy();
        expect(screen.getByTestId('counter').innerHTML).toBe("100");
    });

    test('Debe de mostrar el contador en un h2 con su props pasado', () => {
        const value = 10;
        const { /*container,*/ getByText, getByTestId } = render(<FirstApp value={value} />);
        expect(getByText(value)).toBeTruthy();
        // const h2 = container.querySelector("h2");
        // expect(h2.innerHTML).toContain(JSON.stringify(value));
        expect(getByTestId('counter').innerHTML).toBe(JSON.stringify(value));
    });

    test('Debe de mostrar el contador en un h2', () => {
        render(<FirstApp value={250} />);
        expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe("250");
    });

    test('Debe incrementar con el botÃ³n +1', () => {
        render(<FirstApp value={250} />);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByTestId('counter').innerHTML).toBe("251");
    });

    test('Debe decrementar con el botÃ³n -1', () => {
        render(<FirstApp value={250} />);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByTestId('counter').innerHTML).toBe("249");
    });

    test('Debe reseter su valor por el valor del prop pasado', () => {
        render(<FirstApp value={250} />);
        for (let index = 0; index < 10; index++) {
            fireEvent.click(screen.getByText('+1'));
        }
        // screen.getByRole('button', {name: 'elotes'}); // find elements with role button and name elotes
        fireEvent.click(screen.getByRole('button', { name: 'btn-reset' }));
        expect(screen.getByTestId('counter').innerHTML).toBe("250");
    });

});