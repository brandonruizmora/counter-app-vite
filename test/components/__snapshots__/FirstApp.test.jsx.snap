// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suit First App Debe hacer match con el snapshot estado default 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="center-text"
    >
      <h1>
        CounterApp
      </h1>
      <h2
        data-testid="counter"
      >
        0
      </h2>
      <button
        type="button"
      >
        +1
      </button>
      <button
        type="button"
      >
        -1
      </button>
      <button
        aria-label="btn-reset"
        type="button"
      >
        reset
      </button>
    </div>
  </div>
</div>
`;
